server/controllers/nlp.js

```diff
@@ -132,93 +143,90 @@ function findCommand(state) {
-                conditionMatched = true;
-            } else {
-                _.forEach(columnSynonyms, function (column) {
-                    let firstTokenMatched = _.filter(column.synonyms, function (o) {
-                        let firstWord = o.replace(/ .*/, '');
-                        return firstWord === currentToken;
-                    });
-                    if (firstTokenMatched.length > 0) {
-                        //look if the second matches
-                        _.forEach(firstTokenMatched, function (item) {
-                            let words = item.split(' ');
-                            if (words[1] === state.tokens[state.currentToken + 1]) {
-                                //replace the currentToken
-                                state.tokens[state.currentToken] = {type: staticWords.column, value: column.column_name};
-                                state.tokens.splice(state.currentToken + 1, 1);
-                                conditionMatched = true;
-                            }
-                        });
-                    }
-                });
-            }
-        }
-        nextAction(conditionMatched, state);
+
+        let columnNames = Object.getOwnPropertyNames(columnInfo);
+        let possibleColumns = columnNames.concat(columnSynonyms.map(syn => {
+            return syn.column_name;
+        }));
+        classifyToken(state, staticWords.column, possibleColumns, true);
     });
 }
 
 function extractChartType(state) {
-    let currentToken = state.tokens[state.currentToken];
     let possibleTypes = ["histogram", "pie chart", "line chart", "bar chart", "scatter plot"];
-    let conditionMatched = false;
-    let tokenMatched = _.includes(possibleTypes, currentToken);
+    classifyToken(state, staticWords.chartType, possibleTypes, true);
+}
 
-    if (tokenMatched) {
-        // this layer matches for this token on single word
-        state.tokens[state.currentToken] = {type: staticWords.chartType, value: currentToken};
-        conditionMatched = true;
-    } else {
-        //search if the first word of a Type matches
-        tokenMatched = _.find(possibleTypes, function (o) {
-            let firstWord = o.replace(/ .*/, '');
-            return firstWord === currentToken;
-        });
-        if (tokenMatched) {
-            //look if the second matches
-            let words = tokenMatched.split(' ');
-            if (words[1] === state.tokens[state.currentToken + 1]) {
-                //replace the currentToken
-                state.tokens[state.currentToken] = {type: staticWords.chartType, value: tokenMatched};
-                state.tokens.splice(state.currentToken + 1, 1);
-                conditionMatched = true;
-            }
+//state: current state
+//type: classification category (chartType, column, operation ...)
+//valueRange: possible values of the category
+//lookahead: false if no lookahead should be performed
```

Comment:
could you stick to usual jsDoc?
https://github.com/documentationjs/documentation/blob/master/docs/GETTING_STARTED.md